invenio_github/models.py

```diff
@@ -49,78 +123,160 @@ class Repository(db.Model, Timestamp):
-        repo = cls.get(name, user_id) or Repository(name=name)
-        repo.enabled = enabled
-        repo.user_id = user_id if enabled else None
-        db.session.add(repo)
+    def create(cls, user_id, github_id=None, name=None, **kwargs):
+        """Create the repository."""
+        with db.session.begin_nested():
+            obj = cls(user_id=user_id, github_id=github_id, name=name,
+                      **kwargs)
+            db.session.add(obj)
+        return obj
+
+    @classmethod
+    def get(cls, user_id, github_id=None, name=None):
+        """Return a repository."""
+        repo = cls.query.filter((Repository.github_id == github_id) |
+                                (Repository.name == name)).first()
+        if repo and repo.user_id and str(repo.user_id) != str(user_id):
+            raise Exception('Access forbidden for this repository.')
         return repo
 
     @classmethod
-    def enable(cls, name, user_id):
+    def enable(cls, user_id, github_id, name):
         """Enable webhooks for a repository.
 
-        If the repository does not exits it will create one. Raises 403
+        If the repository does not exist it will create one. Raises 403
         exception if it tries to enable hook for repository created by
         other user.
 
+        :param user_id: User identifier.
+        :param repo_id: GitHub id of the repository.
         :param name: Fully qualified name of the repository.
+        """
+        repo = cls.get_or_create(user_id, github_id=github_id, name=name)
+        repo.enabled = True
+        repo.user_id = user_id
+        db.session.add(repo)
+        return repo
+
+    @classmethod
+    def disable(cls, user_id, github_id, name):
+        """Disable webhooks for a repository.
+
+        Disables the webhook from a repository if its exists in the DB.
+
         :param user_id: User identifier.
+        :param repo_id: GitHub id of the repository.
+        :param name: Fully qualified name of the repository.
         """
-        return cls._change_state(name, user_id, enabled=True)
+        repo = cls.get(user_id, github_id=github_id, name=name)
+        if repo:
+            repo.enabled = False
+            repo.user_id = None
+            repo.hook = None
+            db.session.add(repo)
+        return repo
 
     @classmethod
-    def disable(cls, name, user_id):
-        """Disable webhooks for a repository."""
-        return cls._change_state(name, user_id, enabled=False)
+    def update_ping(cls, repo_id):
+        """Update ping to current time."""
+        repository = cls.query.filter_by(
+            github_id=repo_id,
+            enabled=True,
+        ).one()
+        repository.ping = datetime.utcnow()
+        db.session.commit()
+
+    @property
+    def latest_release(self):
+        """Chronologically latest published release of the repository."""
+        return self.releases.filter_by(
+            status=ReleaseStatus.PUBLISHED).order_by('created desc').first()
```

Comment:
Please try to stick to ORM instead of string values for building queries: `.order_by(db.desc(Repository.created))`